p2p/dhtdiscovery/discovery.go

```diff
@@ -278,5 +281,12 @@ func newDht(ctx context.Context, h host.Host, public, server bool, dir string) (
 	} else {
 		opts = append(opts, dht.Mode(dht.ModeAutoServer))
 	}
-	return dht.New(ctx, h, opts...)
+	dht, err := dht.New(ctx, h, opts...)
+	if err != nil {
+		ds.Close()
```

Comment:
The error from `ds.Close()` is not handled. Perhaps it could be at least logged?

Reply:
I added a log here, since if this error is triggered the node will shut down anyway.